recipes/bison/all/conanfile.py

```diff
@@ -1,70 +1,116 @@
-
 from conans import ConanFile, AutoToolsBuildEnvironment, tools
 from conans.errors import ConanInvalidConfiguration
-import glob
+from contextlib import contextmanager
 import os
-import shutil
 
 
-class ConanFileDefault(ConanFile):
+class BisonConan(ConanFile):
     name = "bison"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://www.gnu.org/software/bison/"
     description = "Bison is a general-purpose parser generator"
     topics = ("conan", "bison", "parser")
     license = "GPL-3.0-or-later"
-    _source_subfolder = "source_subfolder"
-    requires = ("m4/1.4.18")
-    build_requires = ("flex/2.6.4")
```

Comment:
so in the end, flex is not needed *at all* !?

Reply:
flex is only needed when you change the `.l` files in the sources.

Reply:
There are also `.y` files in the sources which need bison. That's why I don't change these because otherwise, bison is needed to regenerate the sources.